rel/marathon.cpp

```diff
@@ -0,0 +1,120 @@
+static void apply_saved_vel() {
+    mkb::Ball& ball = mkb::balls[mkb::curr_player_idx];  // I think this makes ball.vel work
+
+    // Set up the TF matrix for World ---> Start
+    mkb::mtxa_from_rotate_z(mkb::stagedef->start->rotation.z);
+    mkb::mtxa_rotate_y(mkb::stagedef->start->rotation.y);
+    mkb::mtxa_rotate_x(mkb::stagedef->start->rotation.x);
+    // Applies this TF ^^^
+    Vec entered_goal_vel_rt_start = {};
+    mkb::mtxa_tf_vec(&s_saved_vel, &entered_goal_vel_rt_start);
+    // We now have figured out what velocity to apply to the start ball, called
+    // entered_goal_vel_rt_start
+
+    ball.vel = entered_goal_vel_rt_start;  // Apply it to the ball! (finally)
+    // Now that it's used, get rid of the stored vector
+
+    // MATRIX NOTES FROM COMPLEX (THE GOAT) BELOW vvv
+    // Vec pos_rt_itemgroup = {5, 5, 0};
+    // Vec pos_rt_world = {};
+    // mtxa = [ig_to_world]
+    // mkb::mtxa_from_mtx(&mkb::itemgroups[0].transform);
+    // pos_rt_world = mtxa * pos_rt_itemgroup
+    // mkb::mtxa_tf_point(&pos_rt_itemgroup, &pos_rt_world);
+}
+
+// [Nambo] Transforms the current ball velocity relative to the goal, and stores it
+static void store_saved_vel() {
+    mkb::Ball& ball = mkb::balls[mkb::curr_player_idx];  // I think this makes ball.vel work
+
+    // Gives names to the item group index for the goal's IG, and the goal's index inside that IG
+    // This keeps the [] stuff inside the TF code short!
+    s16 goal_idx = mkb::mode_info.entered_goal_idx;
+    s16 goalIG_idx = mkb::mode_info.entered_goal_itemgroup_idx;
+
+    // MATRIX NOTES FROM COMPLEX (THE GOAT) BELOW vvv
+    // Build [world_from_goal]
+    // Build [world_from_itemgroup] * [itemgroup_from_goal]
+    // Build [world_from_itemgroup] * [translation(goal)] * [zrotation(goal)] * [yrotation(goal)] *
+    // [xrotation(goal)]
+
+    // Set up the TF matrix for Goal ---> World (Inverse of World ---> Goal)
+    mkb::mtxa_from_mtx(&mkb::itemgroups[goalIG_idx].transform);  // Build [world_from_itemgroup]
+    mkb::mtxa_translate(&mkb::stagedef->coli_header_list[goalIG_idx].goal_list[goal_idx].position);
+    mkb::mtxa_rotate_z(mkb::stagedef->coli_header_list[goalIG_idx].goal_list[goal_idx].rotation.z);
+    mkb::mtxa_rotate_y(mkb::stagedef->coli_header_list[goalIG_idx].goal_list[goal_idx].rotation.y);
+    mkb::mtxa_rotate_x(mkb::stagedef->coli_header_list[goalIG_idx].goal_list[goal_idx].rotation.x);
+    // Applies this TF ^^^
+    Vec entered_goal_vel_rt_world = {ball.vel.x, ball.vel.y, ball.vel.z};
+    Vec entered_goal_vel_rt_goal = {};
+    mkb::mtxa_rigid_inv_tf_vec(&entered_goal_vel_rt_world, &entered_goal_vel_rt_goal);
+    // We have now stored the desired veloocity vector as entered_goal_vel_rt_goal
+    s_saved_vel = entered_goal_vel_rt_goal;
+}
+
+void tick() {
+    // Mod is turned off
+    if (!pref::get_marathon() && s_state != MarathonState::WaitToStore) {
```

Comment:
I'd remove `&& s_state != MarathonState::WaitToStore`

Reply:
I have it set up this way to prevent the contents from running every frame while the mod is off, instead they just run once & then stop because the state is in WaitToStore. Should I find another way to get this to only run once, or remove the text & leave it running all the time the mod is off?